test/serializationtests.jl

```diff
@@ -0,0 +1,40 @@
+using Serialization
+@testset "Serialize and Deserialize" begin
+
+    cache = LRU{Int, Int}(maxsize=100_000)
+
+    # Populate the cache with dummy data
+    for i in 1:1_000_000
+        cache[i] = i+1
+    end
+    io = IOBuffer()
+    serialize(io, cache)
+    seekstart(io)
+    deserialized_cache = deserialize(io)
+
+    # Check that the cache is the same
+    @test cache.maxsize == deserialized_cache.maxsize
+    @test cache.currentsize == deserialized_cache.currentsize
+    @test cache.hits == deserialized_cache.hits
+    @test cache.misses == deserialized_cache.misses
+    @test cache.by == deserialized_cache.by
+    @test cache.finalizer == deserialized_cache.finalizer
+    @test cache.keyset.length == deserialized_cache.keyset.length
+    @test issetequal(collect(cache), collect(deserialized_cache))
+    # Check that the cache has the same keyset
+    c_node = cache.keyset.first
+    d_node = deserialized_cache.keyset.first
+    for i in 1:cache.keyset.length
+        c_node.val == d_node.val || @test false
```

Comment:
Why not simply `@test c_node.val == d_node.val` ?

Reply:
This is just personal taste, for a large dict of 100k entries, I don't want to add 100k tests that compare each element. I just want to have one test that fails if any element is different, which I believe these do

Reply:
Sure, that is true. On the other hand, the test could also have just been with a handful of elements in the LRU cache I believe. I haven't timed it and believe this is all very fast because it is just `Int`s, but the 100000 did jump to the eye as some large number to have as a simple test case.

Reply:
for small caches, the regular serialization method works fine and doesn't stackoverflow, so we need a big one to really test the new method

Reply:
^ exactly right

Reply:
Well, yes and no, since the new method is now always called, irrespective of the size of the cache. And it is written in such a way that it should not suffer from the same flaws, so I would think that if it passes the test for a smaller cache, that provides sufficient guarantees. But I am fine with the current tests.

Is it clear what was causing the default serialisation strategy to fail? My guess of why it was entering an infinite loop would apply irrespective of the size of the cache, so that is not consistent with the original method working for small caches.